examples/ucp_hello_world.c

```diff
@@ -262,16 +266,22 @@ static int run_ucx_client(ucp_worker_h ucp_worker)
                                   recv_handler);
 
     if (UCS_PTR_IS_ERR(request)) {
-        fprintf(stderr, "unable to receive UCX data message (%u)\n",
-                UCS_PTR_STATUS(request));
-        free(msg);
-        goto err_ep;
+        status = UCS_PTR_STATUS(request);
     } else {
         /* ucp_tag_msg_recv_nb() cannot return NULL */
         assert(UCS_PTR_IS_PTR(request));
         wait(ucp_worker, request);
+        status             = ucp_request_check_status(request);
         request->completed = 0;
         ucp_request_release(request);
+    }
+
+    if (status != UCS_OK) {
+        fprintf(stderr, "unable to receive UCX data message (%s)\n",
+                ucs_status_string(status));
+        free(msg);
```

Comment:
something wrong here: on this lime msg is destroyed via free, but on line 300 - mem_type_free...